airbyte-featureflag/src/main/kotlin/Flags.kt

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.featureflag
+
+/**
+ * If enabled, all messages from the source to the destination will be logged in 1 second intervals.
+ *
+ * This is a permanent flag and would implement the [Flag] type once converted from an environment-variable.
+ */
+object LogConnectorMessages : EnvVar(envVar = "LOG_CONNECTOR_MESSAGES", team = Team.PLATFORM_MOVE)
+
+object StreamCapableState : EnvVar(envVar = "USE_STREAM_CAPABLE_STATE")
+object AutoDetectSchema : EnvVar(envVar = "AUTO_DETECT_SCHEMA", team = Team.PLATFORM_MOVE)
+object NeedStateValidation : EnvVar(envVar = "NEED_STATE_VALIDATION")
+object ApplyFieldSelection : EnvVar(envVar = "APPLY_FIELD_SELECTION")
+
+/**
+ * The team that created and is responsible for individual feature-flags.
+ * TODO: finalize the options here!
+ */
+enum class Team {
+    UNKNOWN,
+    PLATFORM_MOVE,
+    PLATFORM_COMPOSE,
+    FRONTEND,
+}
+
+/**
+ * Flag is a sealed class that all feature-flags must inherit from.
+ *
+ * There are two types of feature-flags; permanent and temporary. Permanent flags should inherit from the Flag class directly
+ * while temporary flags should inherit from the Temporary class (which it itself inherits from the Flag class).
+ *
+ * @param [key] is the globally unique identifier for identifying this specific feature-flag.
+ * @param [default] is the default value of the flag.
+ * @param [team] is the team that is responsible for the feature-flag, defaults to Unknown.
+ */
+sealed class Flag(
+    internal val key: String,
+    internal val default: Boolean = false,
```

Comment:
That's probably fine for now but would it make sense to consider different value types:
```
sealed class Flag<T>(internal val default: T)
```

Reply:
Yes.  

There are different potentials for what a feature-flag could return.  A boolean, an int, a double, a string.  Initially I'm only supporting boolean values as that is all I see the need for.  

Adding additional value types would also change the `FeatureFlagClient` interface to include more than the single `enabled(/*...*/): Boolean)` method.

I went with the simplest solution for now.